src/appleseed/renderer/kernel/lighting/directlightingintegrator.cpp

```diff
@@ -382,8 +382,8 @@ void DirectLightingIntegrator::add_emitting_shape_sample_contribution(
             static_cast<float>(
                 cos_on *
                 rcp_sample_square_distance *
-                edf->get_max_contribution() *
-                sample.m_shape->m_area);
+                sample.m_shape->get_average_flux() *
```

Comment:
Are we still computing `max_contribution` here? Are the variable name and comment up-to-date?

Reply:
@est77 Can we rename `max_contribution` to `shape_flux` ?

Reply:
I don't think using average flux is right. We want the max contribution, i.e. a safe upper bound.

Reply:
I think that the average is a better choice. What happens if you have a mostly black texture with a small high intensity spot?. The light will be probably oversampled.

Reply:
> What happens if you have a mostly black texture with a small high intensity spot?. The light will be probably over-sampled.

It would probably be _correctly_ sampled :) If the texture is black with a high intensity spot, it may well illuminate the entire scene. The idea of this code is to skip shadow rays when we know they aren't useful, not to accidentally miss shadows...

Reply:
We need to write the code to compute this value when using EDFs + textures and OSL emission, we currently do not, so this was not working for those cases anyway.
After that, we can experiment with using max, average or some other heuristic.

Reply:
Sounds good to me, but I suggest introducing a (dummy) `get_max_flux()` method in order to keep this code valid.